sync-core/src/main/java/com/cloudant/http/HttpConnection.java

```diff
@@ -74,6 +77,15 @@
 
     public final HashMap<String, String> requestProperties;
 
+    public final LinkedList<HttpConnectionRequestFilter> requestFilters;
+    public final LinkedList<HttpConnectionResponseFilter> responseFilters;
```

Comment:
Consider making these just `List` objects in the interface, and the requestProperties just a `Map`.